frontend/views/containers/chatroom/ChatMain.vue

```diff
@@ -544,25 +548,27 @@ export default ({
 
       const scrollAndHighlight = (index) => {
         const eleMessage = document.querySelectorAll('.c-body-conversation > .c-message')[index]
-        const eleTarget = document.querySelectorAll('.c-body-conversation > .c-message')[Math.max(0, index - 1)]
 
-        if (!eleTarget) { return }
+        if (!eleMessage) { return }
 
         if (effect) {
-          eleTarget.scrollIntoView({ behavior: this.isReducedMotionMode ? 'instant' : 'smooth' })
+          eleMessage.scrollIntoView({ behavior: this.isReducedMotionMode ? 'instant' : 'smooth' })
           eleMessage.classList.add('c-focused')
           setTimeout(() => {
             eleMessage.classList.remove('c-focused')
           }, 1500)
         } else {
-          eleTarget.scrollIntoView()
+          eleMessage.scrollIntoView()
```

Comment:
This part is the fix for #2548. (It's unclear what the original intention here was but) `eleTarget` here is a message placed above the actual target message and `eleTarget.scrollIntoView(...)` leads to not fully scrolling to the target message. (especially when that previous message has associated file-attachments and takes up large space in the DOM). So made a fix here.